packages/logger/src/winston-logger/logger.ts

```diff
@@ -0,0 +1,169 @@
+import { ConsoleTransport } from '../transports';
+import { ChildLoggerOptions, Log, Logger, LoggerOptions, LogLevel, Transport } from '../types';
+import winston from 'winston';
+import { toWinstonLogLevel, toWinstonTransport } from './adapter';
+import WinstonTransport from 'winston-transport';
+import { format } from 'logform';
+import { nextColor } from './utils';
+import { inspect } from 'util';
+
+const defaultLoggerOptions: LoggerOptions = {
+    transports: [new ConsoleTransport()]
+};
+
+type Metadata = {
+    [key: string]: unknown;
+};
+interface BaseLoggerOptions {
+    logger: winston.Logger;
+    transportMap: Map<Transport, WinstonTransport>;
+    winstonLevel: string;
+    logPrefix: string;
+    metadataOverride?: Metadata;
+}
+
+class BaseWinstonLogger implements Logger {
+    protected _logger: winston.Logger;
+    protected logPrefix: string;
+    protected logPrefixColor: string;
+    protected winstonLevel: string;
+    protected metadataOverride?: Metadata;
+    // Maintain of map of transports. This is useful for adding/removing transports
+    protected transportMap: Map<Transport, WinstonTransport>;
+    // Cache child loggers by logPrefix. Use weak references to avoid causing memory leaks
+    protected childMap: Map<string, WeakRef<Logger>> = new Map();
```

Comment:
It seems WeakRef is only supported from Node 14.6.0+. We need node 12 support for a few more months atleast.

Reply:
Removed the need for `WeakRef`. Fixed in https://github.com/SAP/open-ux-tools/pull/257/commits/c2348bd8d23c4b47495add3e57d3881c8eb5241f